bots/mainbot/castle.js

```diff
@@ -8,379 +8,163 @@ import combat from './combat.js'
-						return self.buildUnit(SPECS.PROPHET, d.x, d.y);
-					}
-					prophetnum++
-				}
-				self.log('Crusaders built are mod ==0 :' + crusadernum)
-				self.log('prophets built are :' + prophetnum)
-
-				if(crusadernum == 5 && prophetnum == 5)
-				{
-					buildflag = 1
-					self.castleTalk(1)
-				}
-
-			}
-		else if (castle_ids[i] % 3 == 1 && robotsnearme[i].castle_talk == 1)
-		{
-			self.log('Inside mod value == 1')
-			if ( self.karbonite > 30 && self.fuel > 150 )
-			   {
-				   var d = getBuildDir(self.me);
-				   if (!(d === undefined)){
-				   self.log('Building a crusader at' + (self.me.x+1) + ',' + (self.me.y+1));
-				 return self.buildUnit(SPECS.CRUSADER, d.x, d.y);
-				   }
-				crusadernum++
-			   }
-			if ( self.karbonite > 30 && self.fuel > 150 )
-			{
-				var d = getBuildDir(self.me);
-				if (!(d === undefined)){
-					self.log('Building a prophet at' + (self.me.x+1) + ',' + (self.me.y+1));
-					return self.buildUnit(SPECS.PROPHET, d.x, d.y);
-				}
-				prophetnum++
-			}
-			self.log('Crusaders built are mod == 1 :' + crusadernum)
-            self.log('prophets built are :' + prophetnum)
-			if(crusadernum == 5 && prophetnum == 5)
-			{
-				buildflag = 1
-				self.castleTalk(1)
-			}
-		}
-		else if(castle_ids[i] % 3 == 2 && robotsnearme[i].castle_talk == 1 )
-		{
-			self.log('Inside mod answer == 2')
-		if ( self.karbonite > 30 && self.fuel > 150 )
-			   {
-				   var d = getBuildDir(self.me);
-				   if (!(d === undefined)){
-				   self.log('Building a crusader at' + (self.me.x+1) + ',' + (self.me.y+1));
-				 return self.buildUnit(SPECS.CRUSADER, d.x, d.y);
-				   }
-				crusadernum++
-			   }
-			if ( self.karbonite > 30 && self.fuel > 150 )
-			{
-				var d = getBuildDir(self.me);
-				if (!(d === undefined)){
-					self.log('Building a prophet at' + (self.me.x+1) + ',' + (self.me.y+1));
-					return self.buildUnit(SPECS.PROPHET, d.x, d.y);
-				}
-				prophetnum++
-			}
-			self.log('Crusaders built are mod ==2 :' + crusadernum)
-			self.log('prophets built are :' + prophetnum)
-			if(crusadernum == 5 && prophetnum == 5)
-			{
-				buildflag = 0
-				self.castleTalk(1)
-			}
-
-		}
-
+		utilities.log(self, `Castle Location: ${[self.me.x, self.me.y]}`)
+		self.castle_count = visible.length;
+		utilities.log(self, `Found ${self.castle_count} castles`);
+        if(self.map.length <= 40 || self.castle_count === 1){
+            self.strategy = 'crusaderRush'
+        }
+        else{
+            self.strategy = 'prophetArmy'
+        }
+        self.builtItems = 0;
+
+        if(self.castle_count === 1){
+            // IF only one castle set the queuePos to 0.
+            // Anything mod the
+            self.queue = [self.id];
+        }
+        else{
+            // Sort by ids
+            let castleIds = []
+            for(let i = 0; i < visible.length; ++i){
+                castleIds.push(visible[i].id)
+            }
+            self.queue = castleIds.sort(function(a, b){return a - b})
```

Comment:
the sort function will sort numbers by default, without needing to explicitly pass a compare function. Not sure if there is a speed boost for using the builtin compare